File: 1518-water-bottles/main.go
func numWaterBottles(numBottles int, numExchange int) int {

    totalDrunk, emptyBottles := numBottles, numBottles

    for emptyBottles >= numExchange {

        newFullBottles := emptyBottles / numExchange
        totalDrunk += newFullBottles

        emptyBottles = emptyBottles % numExchange + newFullBottles
    }   

    return totalDrunk
}

File: 1518-water-bottles/main.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        
        int totalDrunk = numBottles, emptyBottles = numBottles;

        while (emptyBottles >= numExchange) {

            int newFullBottles = emptyBottles / numExchange;
            totalDrunk += newFullBottles;

            emptyBottles = emptyBottles % numExchange + newFullBottles;
        }

        return totalDrunk;
    }
};
